src/python_testing/matter_testing_infrastructure/chip/testing/apps.py

```diff
@@ -102,3 +145,83 @@ def __init__(self, app: str, rpc_server_port: int, storage_dir: str,
         # Start the server application
         super().__init__(*command,  # Pass the constructed command list
                          output_cb=lambda line, is_stderr: self.PREFIX + line)
+
+
+class OTAProviderSubprocess(AppServerSubprocess):
+    """Wrapper class for starting an OTA Provider application server in a subprocess."""
+
+    DEFAULT_ADMIN_NODE_ID = 112233
+
+    # Prefix for log messages from the OTA provider application.
+    PREFIX = b"[OTA-PROVIDER]"
+
+    def __init__(self, app: str, storage_dir: str, discriminator: int,
+                 passcode: int, ota_source: Union[OtaImagePath, ImageListPath],
+                 port: int = 5540, extra_args: list[str] = []):
```

Comment:
suggest using a different port by default - the app will be using 5540 so you'll see port collisions if this is run on the same system.